### DISCUSSION TITLE
RFM98PW SX1278 & Micro SD Card Module Problem

### DISCUSSION THREAD

**Discussion by tolgaeng** [jgromes/RadioLib#356] [Q&A]
Hi Jan!

I've been trying to use Micro SD Card module and LoRa module together. When I try to transmit data from my LoRa without Micro SD Card Module, it works without any problems. But If I integrate my Micro SD Card into PCB, only Micro SD Card Module works and LoRa gives "-2" error code. 

As I know, these library does the CS changes. But I tried to do HIGH, LOW changes in code (Radiolib example codes), again.

I'm using ATMega2560 and here's the some pinouts:

RFM98PW LoRa Module (25, 48, 4, 47) NSS, DIO0, RESET, DIO1
Micro SD Card Module (53) CS
And I've a W25Q64JVSSIQ Flash (10) CS

Thanks

**Comment by jgromes** [comment]
You shouldn't be trying to control the CS pin manually, as it will interfere with the library functions - definitely in the case of RadioLib, likely in the case of any other library. Just don't do it.

I've had a problem like that in the past too, it seems like some SPI devices will just ignore the CS signal and act as if they're the only one on the bus. The only reliable solution I've found is to put them on a separate SPI bus. Since ATmega2560 only has one hardware SPI, I think I used a software-emulated SPI, like this one: https://github.com/MajenkoLibraries/SoftSPI
